drivers/i2c/i2c_ll_stm32.c

```diff
@@ -191,6 +191,8 @@ static void i2c_stm32_irq_config_func_2(struct device *dev)
 
 #endif /* CONFIG_I2C_2 */
 
+#ifndef SOC_SERIES_STM32F1X
```

Comment:
I2C_X availability  depends on SoC and not family.
Use #ifdef I2C3_BASE, defined in ext/hal/st/stm32cube/stm32xxxx/soc/stm32fxxxx.c files, and hence on a SoC version basis, to avoid troubles